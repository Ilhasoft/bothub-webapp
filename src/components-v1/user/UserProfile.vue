<template>
  <div>
    <h4 class="text-color-gray">
      <span>{{ profile.name || profile.nickname }}</span>
      <span>({{ profile.nickname }})</span>
    </h4>
    <p class="text-color-gray">{{ profile.locale || '-' }}</p>
    <p class="text-color-gray">
      <a
        :href="profileURL"
        class="text-color-gray"
        target="_blank">{{ profileURL }}</a>
    </p>
  </div>
</template>

<script>
export default {
  name: 'UserProfile',
  props: {
    profile: {
      type: Object,
      required: true,
    },
  },
  computed: {
    profileURL() {
      return `https://bothub.it/${this.profile.nickname}/`;
    },
  },
};
</script>
