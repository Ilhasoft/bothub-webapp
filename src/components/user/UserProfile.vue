<template>
  <div>
    <h1 class="title is-5 is-marginless">
      <span>{{ profile.name || profile.nickname }}</span>
      <span>({{ profile.nickname }})</span>
    </h1>
    <p>{{ profile.locale || '-' }}</p>
    <p><a
      :href="profileURL"
      target="_blank"
      class="has-text-grey">{{ profileURL }}</a></p>
  </div>
</template>

<script>
export default {
  name: 'UserProfile',
  props: {
    profile: {
      type: Object,
      required: true,
    },
  },
  computed: {
    profileURL() {
      return `https://bothub.it/${this.profile.nickname}/`;
    },
  },
};
</script>

