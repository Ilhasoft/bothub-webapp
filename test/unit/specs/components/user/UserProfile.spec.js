import { shallowMount, createLocalVue } from '@vue/test-utils';
import UserProfile from '@/components/user/UserProfile';


const localVue = createLocalVue();

describe('UserProfile.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(UserProfile, {
      localVue,
      propsData: {
        profile: {
          name: 'Name',
          nickname: 'nickname',
          locale: 'locale',
        },
      },
    });
  });

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
