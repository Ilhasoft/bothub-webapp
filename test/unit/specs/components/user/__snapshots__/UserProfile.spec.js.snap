// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile.vue renders correctly 1`] = `
<div>
  <h4 class="text-color-grey-light"><span>Name</span> <span>(nickname)</span></h4>
  <p class="text-color-grey-light">locale</p>
  <p class="text-color-grey-light"><a href="http://localhost:8080/nickname/" target="_blank" class="text-color-grey-light">http://localhost:8080/nickname/</a></p>
</div>
`;
